workflow/usage.rst

```diff
@@ -450,3 +450,216 @@ The following example shows these functions in action:
     {% if 'waiting_some_approval' in workflow_marked_places(post) %}
         <span class="label">PENDING</span>
     {% endif %}
+
+Storing Metadata
+----------------
+
+.. versionadded:: 4.1
+    The feature to store metadata in workflows was introduced in Symfony 4.1.
+
+In case you need it, you can store arbitrary metadata in workflows, their
+places, and their transitions using the ``metadata`` option. This metadata can
+be as simple as the title of the workflow or as complex as your own application
+requires:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/workflow.yaml
+        framework:
+            workflows:
+                blog_publishing:
+                    metadata: 'Blog Publishing Workflow'
+                    # ...
+                    places:
+                        draft:
+                            metadata:
+                                max_num_of_words: 500
+                        # ...
+                    transitions:
+                        to_review:
+                            from: draft
+                            to:   review
+                            metadata:
+                                priority: 0.5
+                        # ...
+
+    .. code-block:: xml
+
+        <!-- config/packages/workflow.xml -->
+        <?xml version="1.0" encoding="utf-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony http://symfony.com/schema/dic/symfony/symfony-1.0.xsd"
+        >
+
+            <framework:config>
+                <framework:workflow name="blog_publishing" type="workflow">
```

Comment:
You should either remove the `type` here, or add it in other formats (php and yaml) too to remain consistent across examples with different config types.

Reply:
I've removed to make it consistent. There's a discussion to be had around whether we should be explicit about options, or exclude the defaults?

I have not found all possible options documented anywhere, and it would save developers time if they were listed.

Reply:
We encourage doing the following:
```bash
bin/console config:dump-reference framework workflows
```
Do you think we should make it more explicit somewhere in the docs of the integration of the component?

Reply:
> Do you think we should make it more explicit somewhere in the docs of the integration of the component?

Definitely! I didn't know that existed, and I've been using Symfony since 2.4...

Reply:
Added to the documentation in https://github.com/symfony/symfony-docs/pull/11209/commits/60aa8b16342fc1b798a307a4c038bbe148917fd5

Reply:
Thanks!